src/plopp/core/node.py

```diff
@@ -120,11 +120,21 @@ def request_data(self) -> Any:
             self._data = self.func(*args, **kwargs)
         return self._data
 
-    def add_child(self, child: Node):
+    def add_parents(self, *parents: Node):
         """
-        Add a child to the node.
+        Add one or more parents to the node.
         """
-        self.children.append(child)
+        for parent in parents:
+            self.parents.append(parent)
+            parent.children.append(self)
```

Comment:
Is there a risk to add the same parent multiple times? Should the `parents` and `children` attributes be sets instead of lists?

Reply:
Maybe I should do that yes. I will try to see how this looks like in tests.

Reply:
Do you think users may want to control the order in which updates are made? Because we would maybe lose that ability by using sets?
My feeling is that building something that relies on an order sounds flaky and should be avoided anyway, but maybe there are cases where avoiding this is impossible?

Reply:
So I don't think we can use sets, at least not for the `parents`, because they represent the arguments to the function and they should preserve order.

Reply:
So it would also be ok to have the same parent multiple times if you want its output to be passed as multiple function args? Then only the children need to be unique to prevent notifying them multiple times, I guess.

Reply:
I think having the same parent twice would indicate you have messed up somewhere, since in principle there is no reason to have the same value twice as different function arguments?

Unless you are thinking of when writing generic code, you could send the same data as e.g. `signal` and `background` and subtract one from the other?

Reply:
But it would probably result in some strange/useless output in most cases.